File: task-supervise-uniapp/src/api/task.ts
import http from '@/utils/http'

/** 任务状态 */
export type TaskStatus = 'pending' | 'in_progress' | 'pending_feedback' | 'pending_accept' | 'completed' | 'overdue'

/** 任务优先级 */
export type TaskPriority = 'high' | 'medium' | 'low'

/** 任务分派模式 */
export type AssigneeMode = 1 | 2 // 1-单人 2-多人协办

/** 任务信息 */
export interface Task {
  assigneeId?: number
  assigneeName?: string
  assigneeMode?: AssigneeMode
  multiAssigneeIds?: number[]
  attachments?: string[]
  createdAt: string
  creatorId?: number
  creatorName?: string
  deadline?: string
  description?: string
  groupId?: number
  id: number
  taskName?: string
  priority: TaskPriority
  remark?: string
  rejectRemark?: string
  rejectedAt?: string
  acceptResult?: number
  acceptRemark?: string
  acceptedAt?: string
  remainingDays?: number
  overdueDays?: number
  feedbackCount?: number
  status: TaskStatus
  templateId?: number
  title: string
  updatedAt: string
}

/** 任务列表查询参数 */
export interface TaskListParams {
  assigneeId?: number
  keyword?: string
  page?: number
  pageSize?: number
  priority?: TaskPriority
  status?: TaskStatus
}

/** 任务列表响应 */
export interface TaskListResult {
  list: Task[]
  page: number
  pageSize: number
  total: number
}

/** 创建任务参数 */
export interface CreateTaskParams {
  assigneeId?: number
  deadline?: string
  description?: string
  priority?: TaskPriority
  remark?: string
  title: string
}

/** 更新任务参数 */
export interface UpdateTaskParams extends Partial<CreateTaskParams> {
  status?: TaskStatus
}

/** 任务模板信息 */
export interface TaskTemplate {
  templateId: number
  templateName: string
  templateType: number
  description?: string
  defaultContent?: string
  defaultPriority?: number
  standardFeedbackReq?: string
  groupId?: number
  status: number
}

/** 任务指派人信息 */
export interface TaskAssignee {
  assigneeId: number
  taskId: number
  userId: number
  assigneeName: string
  assigneeType: number // 1-主负责人 2-协助人
  status: string
}

/** 任务 API */
export const taskApi = {
  /** 指派任务（单人模式） */
  assign: (id: number, assigneeId: number) =>
    http.post(`/task/${id}/assign`, { assigneeId }),

  /** 多人协办模式分派任务 */
  assignMulti: (taskId: number, assigneeIds: number[], primaryId?: number) =>
    http.post('/task/assign-multi', { taskId, assigneeIds, primaryId }),

  /** 批量分派任务 */
  batchAssign: (taskIds: number[], assigneeId: number) =>
    http.post('/task/batch-assign', { taskIds, assigneeId }),

  /** 根据模板创建任务 */
  createByTemplate: (templateId: number, params?: Partial<CreateTaskParams>) =>
    http.post('/task/create-by-template', { templateId, ...params }),

  /** 创建任务 */
  create: (params: CreateTaskParams) => http.post<Task>('/task', params),

  /** 删除任务 */
  delete: (id: number) => http.delete(`/task/${id}`),

  /** 获取任务详情 */
  getDetail: (id: number) => http.get<Task>(`/task/${id}`),

  /** 获取任务列表 */
  getList: (params?: TaskListParams) =>
    http.get<TaskListResult>('/task/list', params),

  /** 按分组查询任务列表 */
  getListByGroup: (groupId: number, page?: number, pageSize?: number) =>
    http.get<TaskListResult>(`/task/group/${groupId}`, { page, pageSize }),

  /** 获取我的任务统计 */
  getStatistics: () => http.get('/task/statistics'),

  /** 更新任务 */
  update: (id: number, params: UpdateTaskParams) =>
    http.put<Task>(`/task/${id}`, params),

  /** 更新任务状态 */
  updateStatus: (id: number, status: TaskStatus) =>
    http.put(`/task/${id}/status`, { status }),

  // ========== 模板相关接口 ==========

  /** 获取启用的模板列表 */
  getEnabledTemplates: (templateType?: number) =>
    http.get<TaskTemplate[]>('/task/template/enabled', { templateType }),

  /** 获取模板详情 */
  getTemplateDetail: (templateId: number) =>
    http.get<TaskTemplate>(`/task/template/${templateId}`),

  /** 获取模板统计 */
  getTemplateStatistics: () =>
    http.get('/task/template/statistics'),

  /** 驳回已完成的任务（管理员/主管） */
  reject: (id: number, rejectRemark: string) =>
    http.put(`/task/${id}/reject`, { rejectRemark }),

  /** 验收任务（管理员/主管） */
  accept: (id: number, acceptResult: number, acceptRemark: string) =>
    http.put(`/task/${id}/accept`, { acceptResult, acceptRemark }),

  /** 获取任务全流程时间线 */
  getTimeline: (id: number) =>
    http.get(`/task/${id}/timeline`),

  /** 获取逾期任务列表 */
  overdueList: (params?: { page?: number; limit?: number }) =>
    http.get('/task/overdue/list', params),

  /** 批量处置逾期任务 */
  batchOverdueAction: (taskIds: string[], action: string, remark: string) =>
    http.post('/task/batch-overdue-action', { taskIds, action, remark }),
}

/** 任务指派人 API */
export const taskAssigneeApi = {
  /** 获取任务指派人列表 */
  listByTaskId: (taskId: number) =>
    http.get<TaskAssignee[]>(`/task/assignee/task/${taskId}`),

  /** 批量添加指派人 */
  batchCreate: (taskId: number, userIds: number[], assigneeType?: number) =>
    http.post('/task/assignee/batch', { taskId, userIds, assigneeType }),

  /** 更新指派类型 */
  updateAssigneeType: (id: number, assigneeType: number) =>
    http.put(`/task/assignee/${id}/type`, { assigneeType }),

  /** 按任务ID和指派类型查询指派人列表 */
  listByTaskIdAndType: (taskId: number, assigneeType: number) =>
    http.get<TaskAssignee[]>(`/task/assignee/task/${taskId}/type/${assigneeType}`),
}

export default taskApi

File: task-supervise-server/src/main/java/com/enterprise/tasksuperviseserver/module/task/entity/Task.java
package com.enterprise.tasksuperviseserver.module.task.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.experimental.Accessors;
import java.time.LocalDateTime;

/**
 * 任务实体
 * <p>
 * 映射表 task: id, title, description, status, priority, deadline, remark, attachments,
 * creator_id, creator_name, assignee_id, assignee_name, deleted, created_at, updated_at
 *
 * @author grq
 * @date 2026-08-26
 * @version v1.0.0
 */
@Data
@Accessors(chain = true)
@TableName("task")
public class Task {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    private String title;

    private String description;

    /** 状态: pending/in_progress/completed/overdue */
    private String status;

    /** 优先级: high/medium/low */
    private String priority;

    /** 所属部门ID，用于权限过滤 */
    @TableField("dept_id")
    private Long deptId;

    /** 关联模板ID */
    @TableField("template_id")
    private Long templateId;

    /** 分派模式: 1-单人 2-多人协办 */
    @TableField("assignee_mode")
    private Integer assigneeMode;

    /** 任务组ID */
    @TableField("group_id")
    private Long groupId;

    private LocalDateTime deadline;

    private String remark;

    /** 驳回原因（管理员/主管驳回时填写） */
    @TableField("reject_remark")
    private String rejectRemark;

    /** 驳回时间 */
    @TableField("rejected_at")
    private LocalDateTime rejectedAt;

    /** 验收结果: 0待验收 1通过 2驳回 */
    @TableField("accept_result")
    private Integer acceptResult;

    /** 验收意见 */
    @TableField("accept_remark")
    private String acceptRemark;

    /** 验收时间 */
    @TableField("accepted_at")
    private LocalDateTime acceptedAt;

    /** 验收人ID */
    @TableField("accepted_by")
    private Long acceptedBy;

    private String attachments;

    @TableField("creator_id")
    private Long creatorId;

    @TableField("creator_name")
    private String creatorName;

    @TableField("assignee_id")
    private Long assigneeId;

    @TableField("assignee_name")
    private String assigneeName;

    @TableLogic
    private Integer deleted;

    /** 逾期标记: 0-未标记 1-已标记（防止重复触发逾期提醒） */
    @TableField("overdue_marked")
    private Integer overdueMarked;

    @TableField("created_at")
    private LocalDateTime createdAt;

    @TableField("updated_at")
    private LocalDateTime updatedAt;

}
